api/main.py

```diff
@@ -90,6 +90,28 @@ async def get_nodes(request: Request):
     return await db.find_by_attributes(Node, dict(request.query_params))
 
 
+@app.get('/get_root_node/{node_id}', response_model=Node)
+async def get_root_node(node_id: str):
+    try:
+        while node_id:
+            node = await db.find_by_id(Node, node_id)
+            if node is None:
+                raise ValueError('Node not found with id: {}'.format(node_id))
+            node_id = node.parent
+        return node
+
+    except errors.InvalidId as error:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(error)
+        )
+    except ValueError as error:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(error)
+        )
```

Comment:
I'd rather limit the number of lines in the `try..except` block to the very minimum. 
I think there is no need to `raise` the `ValueError` and catch it immediately to raise the HTTPException. I think we can just raise the latter in case of an error.

```suggestion
        while node_id:
            try:
                node = await db.find_by_id(Node, node_id)
            except errors.InvalidId as error:
                raise HTTPException(
                    status_code=status.HTTP_400_BAD_REQUEST,
                    detail=str(error)
                )
            
            if node is None:
                raise HTTPException(
                    status_code=status.HTTP_400_BAD_REQUEST,
                    detail=('Node not found with id: {}'.format(node_id)
                )
            node_id = node.parent
        return node
```